### PR TITLE
DOC clarify inplace operation section in 1.5 whats_new

### PR DESCRIPTION
**Pull request by lesteve**
xref https://github.com/pandas-dev/pandas/issues/47381

Coming from scikit-learn tests failing on scikit-learn with the pandas development version, I found the whats_new entry not very helpful at all. This does a few things:
- use an example with different dtypes so that the behaviour is as advertised for old behaviour (i.e. `ser` is not updated in place). This was not the case before, since when dtype matches, `ser` is updated in place so the code snippet was not showing the right behaviour.
- make it clearer that there is no behaviour change in pandas 1.5

Some of the wording may be not completely accurate, as I don't have a very good grasp of the pandas internals, feel free to suggest improvements!

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
this needs some editing

i

**Review by jreback — CHANGES_REQUESTED** [review]
i don't know if this is an improvement. can you try to make a minimal change instead.

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -565,37 +565,50 @@ retained by specifying ``group_keys=False``.
...
    df = pd.DataFrame(values)
    ser = df[0]
 
-*Old behavior*:
+*Old behavior with pandas < 1.5*:
 
 .. code-block:: ipython
 
-    In [3]: df.iloc[:, 0] = np.array([10, 11])
+    In [3]: df.iloc[:, 0] = np.array([10, 11], dtype=np.int32)
```
[the remark this answers is not in the document]
Comment (lesteve) [reply doc/source/whatsnew/v1.5.0.rst:585]: When the dtypes match the old (pandas < 1.5) behaviour is already to update the underlying array in place.

```
import numpy as np
import pandas as pd

print(f"{pd.__version__=}")
values = np.arange(4).reshape(2, 2)
df = pd.DataFrame(values)
ser = df[0]
print(f"before setting column\n{ser}")
df.iloc[:, 0] = np.array([10, 11])
print(f"after setting column\n{ser}")
```
Output:
```
pd.__version__='1.4.2'
before setting column
0    0
1    2
Name: 0, dtype: int64
after setting column
0    10
1    11
Name: 0, dtype: int64
```

This snippet was incorrect in pretending that `ser` was not updated in place.

I used a different dtype for the assignment rhs term to be in the case, where:
- the old (pandas < 1.5) behaviour is not to update in place
- the new (pandas 1.5) behaviour is the same with an additional warning
- the future behaviour will be to update in place

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -562,14 +562,15 @@ As ``group_keys=True`` is the default value of :meth:`DataFrame.groupby` and
 raise a ``FutureWarning``. This can be silenced and the previous behavior
 retained by specifying ``group_keys=False``.
 
-.. _whatsnew_150.notable_bug_fixes.setitem_column_try_inplace:
+.. _whatsnew_150.deprecations.setitem_column_try_inplace:
    _ see also _whatsnew_130.notable_bug_fixes.setitem_column_try_inplace
```
Comment (lesteve) [inline doc/source/whatsnew/v1.5.0.rst:566]: side-comment: this is not rendered in the HTML, not sure whether it was supposed to be a link to  https://pandas.pydata.org/docs/dev/whatsnew/v1.3.0.html#try-operating-inplace-when-setting-values-with-loc-and-iloc

**Comment by lesteve** [comment]
I have tried to make the diff smaller as advised

**Review by datapythonista — COMMENTED** [review]
lgtm, added couple of small things, but this makes sense.

If you want to spend the time, I think it'd help read this section is the example had some more "interesting" data. I think it's more difficult to understand the code with the original arbitrary code, and even more now, that one needs to understand what's going on with the types.

Maybe something like:
```python
pandas.DataFrame({'price': [10, 15]}, index=['book1', 'boo2'])
new_prices = [11.50, 18.20]
```
Does this make sense?

**Comment by lesteve** [comment]
> If you want to spend the time, I think it'd help read this section is the example had some more "interesting" data. 

I did this in my last commit, I also showed `df.iloc[:, 0]` which I think helps in understanding what is happening (whether the column keeps its dtype or not).

**Review by datapythonista — APPROVED** [review]
This seems super clear to me now, thanks a lot @lesteve

File: doc/source/whatsnew/v1.5.0.rst (hunk 2 above)
Comment (mroeschke) [reply doc/source/whatsnew/v1.5.0.rst:566]: Yeah it looks like it's supposed to reference that section. Mind correcting it? Can be in the body of the paragraph too like ``` Please reference `this section <whatsnew_130.notable_bug_fixes.setitem_column_try_inplace>` of the 1.3 whatsnew file. ```

**Review by jreback — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lesteve

**Pull request merged**